### DISCUSSION TITLE
Amharic - አማርኛ  (am-et) language support

### DISCUSSION THREAD

**Discussion by yosiasz** [explosion/spaCy#6556] [Language Support]
Greetings

How do you I go about creating a new space language pack?  Would like  to contribute if possible.

Thanks!

**Comment by svlandeg** [comment]
Hi! 

We're always happy to receive community contributions for languages that we don't speak ourselves. You can consult https://spacy.io/usage/adding-languages for more background information on how language data is structured in spaCy.

In [this thread](https://github.com/explosion/spaCy/discussions/3056), you can find more on the different ways you can contribute. For an entirely new language, it would probably be best to focus on correct [tokenization](https://spacy.io/usage/linguistic-features#section-tokenization) first.

**Reply by yosiasz** [reply]
Would have been nice if there was a  lightweight language template. barrier to entry is high

**Reply by yosiasz** [reply]
@svlandeg forked, created some basic Amharic language bits. What are next steps to testing it?

**Reply by svlandeg** [reply]
Did you check the guidelines at https://spacy.io/usage/adding-languages#testing ?

**Reply by yosiasz** [reply]
@svlandeg Awesome! tests pass for Amharic. Now my question is about punctuations. Both Amharic and Tigrinya use their own unique punctuations. Do I add these in ./spacy/lang/char_classes

```
_punct = (
    r"… …… , : ; \! \? ¿ ؟ ¡ \( \) \[ \] \{ \} < > _ # \* & 。 ？ ！ ， 、 ； ： ～ · । ، ۔ ؛ ٪"
)
```

or elsewhere in the language's own punctuation file?

Thanks for the guidance!

**Reply by yosiasz** [reply]
@svlandeg never mind, I did the following

```
_list_punct = LIST_PUNCT + "፡ ። ፣ ፤ ፥ ፦ ፧".strip().split()
```

**Comment by tewodrosm** [comment]
hey, I am adding Amharic language in spacy locally following your guide [Spacy Documentation](https://spacy.io/usage/adding-languages) , how I can contribute to the project. thanks in advance

**Reply by yosiasz** [reply]
@tewodrosm Any progress on this? I am not seeing Amharic in the spaCy language space. thanks!

**Comment by yosiasz** [comment]
this is something interesting

https://arxiv.org/pdf/1710.00923.pdf

**Reply by tewodrosm** [reply]
thanks for sharing [yosiasz](https://github.com/yosiasz).

**Reply by yosiasz** [reply]
@tewodrosm I have a pull request up. https://github.com/explosion/spaCy/pull/6583/commits in case you are interested.
